homeassistant/components/surepetcare/__init__.py

```diff
@@ -198,23 +198,22 @@ async def handle_set_lock_state(self, call: ServiceCall) -> None:
         await self.lock_states[state](flap_id)
         await self.async_request_refresh()
 
-    def get_pets(self) -> dict[int, str]:
+    def get_pets(self) -> dict[str, int]:
         """Get pets."""
         names = {}
         for surepy_entity in self.data.values():
             if surepy_entity.type == EntityType.PET and surepy_entity.name:
-                names[surepy_entity.id] = surepy_entity.name
+                names[surepy_entity.name] = surepy_entity.id
         return names
 
     async def handle_set_pet_location(self, call: ServiceCall) -> None:
         """Call when setting the pet location."""
         pet_name = call.data[ATTR_PET_NAME]
         location = call.data[ATTR_LOCATION]
-        for device_id, device_name in self.get_pets().items():
-            if pet_name == device_name:
-                await self.surepy.sac.set_pet_location(
-                    device_id, Location[location.upper()]
-                )
-                await self.async_request_refresh()
-                return
+        if device_id := self.get_pets().get(pet_name):
```

Comment:
I'd invert this check and raise a ValueError with the error message as exception argument. Then we can outdent below and have more of the code outdented.

Reply:
Can `device_id` be 0?